#Main Dimensions of Stator Core

\begin{equation}
  D_{is} = \sqrt[3]{\frac{2p*p_1*S_{gap}}{pi*\lambda*f_1*C_0}}
\end{equation}

In [189]:
P_n = 150 
V_1 = 1000
n_1 = 1800 
p   = 2
p_1 = 2
f_1 = 50
cos_theta = 0.85
K_E = 0.98 - 0.005*p_1
eff = 0.9
q=3


\begin{equation}
  S_{gap} = \frac{K_E*P_n}{\eta*\cos(\theta_n)}
\end{equation}

In [190]:
print K_E

0.97


In [191]:
S_gap = ((K_E*150*1000)/(eff*cos_theta))
print S_gap

190196.078431


###for a 4 pole machine,
\begin{equation}
\lambda = 1.1
\end{equation}

In [192]:
import math
D_is = math.pow((2*p*p_1*S_gap)/(3.14*1.1*f_1*265000),1.0/3.0)
print D_is

0.321551923037


\begin{equation}
D_{is} = 0.32 m
\end{equation}

\begin{equation}
\tau = \frac{pi*D_{is}}{2*p}
\\\tau_{s} = \frac{\tau}{3*q}
\end{equation}

In [193]:
pole_pitch = 3.14*D_is/(2*p)
print pole_pitch

0.252418259584


\begin{equation}
\tau = 0.25 m
\end{equation}

In [194]:
slot_per_pole= pole_pitch/(3*q)
print slot_per_pole

0.0280464732871


####again for a 4 pole machine;
\begin{equation}
\frac{D_{is}}{D_{out}} = K_D = 0.62
\end{equation}

In [196]:
D_out = D_is/0.62
print D_out

0.51863213393


In [197]:
D_out=0.53


\begin{equation}
D_{out} = 0.52 m
\end{equation}

\begin{equation}
g = (0.1 + 0.012*\sqrt[3]{P_n}*10^{-3})
\\for \quad 2p>2
\end{equation}

In [199]:
import math
g = (0.1+0.012*math.pow(P_n*1000,1.0/3.0))/(1000)
print g

0.00073759514151


\begin{equation}
g = 0.7 mm
\end{equation}

#The Stator Winding 

\begin{equation}
N_s = 2*p*q*m = 2*2*3*3 = 36
\end{equation}

\begin{equation}
\alpha_{ec} = \frac{2\pi*p}{N_s} = \frac{\pi}{9}
\end{equation}

\begin{equation}
K_{ql} = \frac{\sin(\frac{\pi}{6})}{q*\sin(\frac{\pi}{6q})}
\end{equation}

In [137]:
import math
K_ql = math.sin(3.14/6)/(3*math.sin(3.14/18))
print K_ql

0.959835420315


\begin{equation}
for\quad chording : \frac{y}{\tau} = \frac{7}{9}\quad is \quad selected
\end{equation}

\begin{equation}
so\quad the\quad electrical\quad angle\quad is:\quad \frac{2*\pi*p}{N_s}=\quad \frac{\pi}{9}
\end{equation}

\begin{equation}
chording\quad factor\quad is:\quad \sin(\frac{pi*y}{2*\tau})
\end{equation}

In [200]:
import math
K_yl = math.sin((3.14*7)/(2*9))
print K_yl

0.939480605157


In [201]:
import math
Kw1= K_yl*K_ql
print Kw1

0.901746761529


####following values are assumed from now on:


In [202]:
B_g = 0.7 
alpha_i = 0.65
L=1.1*3.14*D_is/(2*p)
rated_current = 265000/(1.1*alpha_i*Kw1*3.14*B_g*p)
print rated_current/1000


93.4969743854


In [203]:
pole_flux=alpha_i*pole_pitch*L*B_g
print pole_flux

0.0318893463746


####number of turns per phase W1:
\begin{equation}
\frac{K_E*V_1}{4*K_f*K_{W1}*f*\Phi}
\end{equation}

In [204]:
import math
W_1 = (K_E*V_1)/(4*1.1*Kw1*50*pole_flux*math.sqrt(3))
print W_1 

88.5234267262


####number of conductors ns:
\begin{equation}
ns:\quad \frac{W_1}{p*q}
\end{equation}
#### (current paths in parallel:1)

In [205]:
ns = W_1/(p*q)
print ns

14.7539044544


In [206]:
ns = 14

In [207]:
import math
rated_current= 150000/(eff*cos_theta*math.sqrt(3)*V_1)
print rated_current


113.205935135


In [146]:
J_co=6.3

####magnetic wire cross section:
\begin{equation}
A_{co}=\quad \frac{I_{rated}}{J_{cos}}
\end{equation}

In [208]:
A_co=rated_current/J_co
print A_co

17.9691960532


\begin{equation}
A_{co}=conductor\quad width*conductor\quad height
\end{equation}
####a rectengular cross section conductor will be used,so standart value of 10 mm is taken here for slot width

In [209]:
slot_width = 10
conductor_insulation = 0.4
slot_insulation = 4
conductor_width = slot_width - conductor_insulation - slot_insulation
print conductor_width

5.6


#### conductor height = A_co/conductor width = 18/5.6 = 3.21mm

In [210]:
radial_cond_insulation = conductor_insulation*ns
radial_slot_insulation = slot_insulation*2
interlayer_insulation = 2
wedge = 4
radial_insulation = radial_cond_insulation + radial_slot_insulation+interlayer_insulation+wedge
print radial_insulation

19.6


####slot height:
\begin{equation}
h_s = h_{ins} + n_s*b_c
\end{equation}

In [211]:
h_s = radial_insulation + ns*3.21
print h_s

64.54


####back iron radial thickness:
\begin{equation}
h_{cs} = \frac{D_{out}-D_{is}}{2}-h_s
\end{equation}

In [212]:
h_cs= ((D_out-D_is)*1000)/2-h_s
print h_cs

39.6840384816


####so all in all;
\begin{equation}
h_s = 64.54 mm
\\h_{cs} = 39.7 mm
\end{equation}
####thus back iron flux density is:
\begin{equation}
B_{cs} = \frac{\Phi}{2*l_i*h_cs}
\end{equation}
####where l_i is stack length, and calculated as 0.277mm

In [213]:
B_cs = pole_flux/(2*L*0.001*h_cs)
print B_cs

1.44705922715


#### 1.8 T is a little big value, so D_out is changed to 0.53m and recalculations are made to obtain 1.44 T

#Rotor Slots

#### Ns was selected as 36 previously, so among the adequate combinations for 2p = 4 and Ns = 36, Nr, the skewed rotor slot number is chosen as 28

In [214]:
Nr=28

In [215]:
import math
Kw2 = math.sin(3.14/6)/(2.3*math.sin(3.14/14))
print Kw2

0.976985413207


\begin{equation}
\tau_r = \frac{\pi(D_{is}-2g)}{N_r}
\end{equation}

In [217]:
rotor_tooth_pitch = 3.14*(D_is*1000-2*g)/(Nr)
print rotor_tooth_pitch

36.0595859371


In [218]:
nr=2
W_2 = Nr*nr/(2*q*1)
print W_2

9


In [219]:
V_2 = V_1*(K_E*W_2*Kw2)/(W_1*Kw1)
print V_2

106.846306751


In [220]:
Ki= 0.936
import math
rated_current_2= (Ki*W_1*Kw1*rated_current)/(W_2*Kw2*math.sqrt(3))
print rated_current_2

555.388185846


In [221]:
j_cor = 7
A_cor = rated_current_2/(j_cor)
print A_cor

79.3411694066


####the slot width b_r is:
\begin{equation}
b_r<\tau_r(1-\frac{B_g}{B_{trmin}})
\end{equation}

In [222]:
Btrmin = (slot_per_pole*1000*B_g)/(slot_per_pole*1000-slot_width)
print Btrmin

1.08788742203


In [223]:
rotor_tooth_pitch*(1-B_g/Btrmin)

12.85708387216765

In [257]:
br=12

#### so 12 can be chosen as the slot width

####considering the insulation width is 2 mm, conductor allowable width is 8.5mm. For 2 vertical conductors,each conductor has 4mm of width. So the height is:

In [258]:
b_c = A_cor/(2*4)
print b_c

9.91764617582


####Rotor back iron height is:
\begin{equation}
h_{cr}=\frac{B_g*\tau*1}{B_{cr}*\pi*K_{Fe}}
\end{equation}


In [259]:
h_cr=(B_g*pole_pitch*1)/(1.4*3.14*0.98)
print h_cr

0.0410142758976


#Rotor performance calculation

####magnetization mmfs:

\begin{equation}
F_{lm}=F_g+F_s+(F_{st}+F_{rt})+\frac{F_{cs}}{2}+\frac{F_{cr}}{2}
\end{equation}

\begin{equation}
F_g=K_c*g*\frac{B_g}{\mu_o}
\\K_c=K_{c1}*K_{c2}
\\K_{c1,c2}=\frac{\tau_{s,r}}{\tau_{s,r}-\gamma_{s,r}g}
\\gamma_{s,r}=\frac{\frac{b_{os,r}}{g}^2}{5+\frac{b_{os,r}}{g}}
\end{equation}

In [260]:
import math
gamma=math.pow(slot_width,2)/(5*g*1000+slot_width)
print gamma


7.30568216488


In [261]:
Kc2= rotor_tooth_pitch/(rotor_tooth_pitch - gamma*g)
print Kc2

1.00014945932


In [262]:
Kc1=1.3

In [229]:
Kc=Kc1*Kc2
print Kc

1.30019429711


In [263]:
import math
Fg = (Kc*g*B_g)/(1.256*math.pow(10,-6))
print Fg

534.483994902


In [297]:
h_st=40
Ns=36
import math
Btsmin=math.pow(3.14*(D_is*1000+2*h_st*2)-Ns*slot_width,-1)*B_g*3.14*D_is*1000
print Btsmin

0.613477707851


In [325]:
h_rt=25
import math
Btrmax=3.14*(D_is*1000-2*g*1000)*B_g/(3.14*(D_is*1000-2*g*1000-2*h_rt)-Nr*br)
print Btrmax

1.37396954515


In [294]:
Btsmed=(3.14*D_is*1000*B_g)/((3.14*(D_is*1000+h_st)-Ns*slot_width))
print Btsmed

0.911641566115


In [293]:

Btrmed=(3.14*D_is*1000*B_g)/(3.14*(D_is*1000-h_rt)-Nr*br)
print Btrmed

1.18750528218


####from lamihation magnetization curve;
Hstmin=106 A/m, Hstmed = 177 A/m, Hstmax= 585 A/m
####and also;
\begin{equation}
F_{st}=h_{st}*H_{st}
\end{equation}

In [330]:
Hts=(106+4*177+585)/6
print Hts

233


In [331]:
Fts=h_st*0.001*Hst
print Fts

9.32


In [332]:
Btrmin = 1.15
Htrmin=310
Htrmed=356
Htrmax=760
Htr=(Hrtmin+4*Hrtmed+Hrtmax)/6
print Htr

415


In [329]:
Ftr=Htr*h_rt*0.001
print Ftr

10.375
